x-pack/plugins/cases/public/components/all_cases/index.tsx

```diff
@@ -0,0 +1,607 @@
+        if (sort) {
+          newQueryParams = {
+            ...newQueryParams,
+            sortField: getSortField(sort.field),
+            sortOrder: sort.direction,
+          };
+        }
+        if (page) {
+          newQueryParams = {
+            ...newQueryParams,
+            page: page.index + 1,
+            perPage: page.size,
+          };
+        }
+        setQueryParams(newQueryParams);
+        refreshCases(false);
+      },
+      [queryParams, refreshCases, setQueryParams]
+    );
+
+    const onFilterChangedCallback = useCallback(
+      (newFilterOptions: Partial<FilterOptions>) => {
+        if (newFilterOptions.status && newFilterOptions.status === CaseStatuses.closed) {
+          setQueryParams({ sortField: SortFieldCase.closedAt });
+        } else if (newFilterOptions.status && newFilterOptions.status === CaseStatuses.open) {
+          setQueryParams({ sortField: SortFieldCase.createdAt });
+        } else if (
+          newFilterOptions.status &&
+          newFilterOptions.status === CaseStatuses['in-progress']
+        ) {
+          setQueryParams({ sortField: SortFieldCase.updatedAt });
+        }
+        setFilters(newFilterOptions);
+        refreshCases(false);
+      },
+      [refreshCases, setQueryParams, setFilters]
+    );
+
+    const memoizedGetCasesColumns = useMemo(
+      () =>
+        getCasesColumns({
+          actions: userCanCrud ? actions : [],
+          filterStatus: filterOptions.status,
+          isModal,
+          getCaseDetailsHref,
+          onCaseDetailsNavClick,
+        }),
+      [
+        userCanCrud,
+        actions,
+        filterOptions.status,
+        isModal,
+        getCaseDetailsHref,
+        onCaseDetailsNavClick,
+      ]
+    );
+
+    const itemIdToExpandedRowMap = useMemo(
+      () =>
+        getExpandedRowMap({
+          columns: memoizedGetCasesColumns,
+          data: data.cases,
+          isModal,
+          onSubCaseClick: onRowClick,
+        }),
+      [data.cases, isModal, memoizedGetCasesColumns, onRowClick]
+    );
+
+    const memoizedPagination = useMemo(
+      () => ({
+        pageIndex: queryParams.page - 1,
+        pageSize: queryParams.perPage,
+        totalItemCount: data.total,
+        pageSizeOptions: [5, 10, 15, 20, 25],
+      }),
+      [data, queryParams]
+    );
+
+    const sorting: EuiTableSortingType<Case> = {
+      sort: { field: queryParams.sortField, direction: queryParams.sortOrder },
+    };
+
+    const euiBasicTableSelectionProps = useMemo<EuiTableSelectionType<Case>>(
+      () => ({
+        onSelectionChange: setSelectedCases,
+        selectableMessage: (selectable) => (!selectable ? SELECTABLE_MESSAGE_COLLECTIONS : ''),
+      }),
+      [setSelectedCases]
+    );
+    const isCasesLoading = useMemo(
+      () => loading.indexOf('cases') > -1 || loading.indexOf('caseUpdate') > -1,
+      [loading]
+    );
+    const isDataEmpty = useMemo(() => data.total === 0, [data]);
```

Comment:
I know this is copy/paste, but a lot of `useMemo`s in this file and many others we are copying. Considering [what we learned recently](https://canimerge.com/should-you-really-use-usememo-in-react-lets-find-out/), I'm thinking we should remove most of these. Maybe it would be worth it for `memoizedGetCasesColumns`, stuff that rerenders a lot. what do you think @XavierM? Can we please agree on when to `useMemo` and when to not? Once we decide that, we could clean these up in a follow up refactor PR, I know I pulled in a lot of `useMemo`s in [my PR](#94444) as well

Reply:
to me as right now, if you are not going to use it as a prop, there is not need of `useMemo` or `useCallback`

Reply:
I've gotten competing opinions about how to handle it from different people, but I personally liked Kent Dodd's thoughts on it: https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
lets handle in a follow up PR for sure